uber/badge_funcs.py

```diff
@@ -98,3 +98,71 @@ def check_unassigned():
 
 
 # TODO: perhaps a check_leaderless() for checking for leaderless groups, since those don't get emails
+
+# run through all badges and check 2 things:
+# 1) there are no gaps in badge numbers
+# 2) all badge numbers are in the ranges set by BADGE_RANGES
+# note: does not do any duplicates checking, that's a different pre-existing check
+def badge_consistency_check(session):
+    errors = []
+
+    # check 1, see if anything is out of range, or has a duplicate badge number
+    badge_nums_seen = []
+    for attendee in session.query(Attendee)\
+            .filter(Attendee.first_name != '', Attendee.badge_num != 0)\
+            .order_by(Attendee.badge_num).all():
+        out_of_range_error = check_range(attendee.badge_num, attendee.badge_type)
+        if out_of_range_error:
+            msg = '{a.full_name}: badge #{a.badge_num}: {err}'.format(a=attendee, err=out_of_range_error)
+            errors.append(msg)
+
+        if attendee.badge_num in badge_nums_seen:
+            msg = '{a.full_name}: badge #{a.badge_num}: Has been assigned the same badge number of another badge, which is not supposed to happen'.format(a=attendee)
+            errors.append(msg)
+
+        badge_nums_seen.append(attendee.badge_num)
```

Comment:
I've just created https://github.com/magfest/ubersystem/issues/1637 to track a more permanent solution to this check.